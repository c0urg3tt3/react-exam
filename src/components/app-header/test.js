import React from 'react'
import { shallow } from 'enzyme'

import AppHeader from './component'

describe('[component] AppHeader', () => {
  it('should render component', () => {
    const wrapper = shallow(<AppHeader />)

    expect(wrapper.exists()).toEqual(true)
    expect(wrapper.is('header')).toEqual(true)
    expect(wrapper.hasClass('app-header')).toEqual(true)
    expect(wrapper.html()).toMatchSnapshot()
  })

  it('should render child', () => {
    const wrapper = shallow(
      <AppHeader>
        <h1>Jedi List</h1>
      </AppHeader>
    )

    expect(wrapper.find('h1').text()).toEqual("Jedi List")
    expect(wrapper.html()).toMatchSnapshot()
  })

  it('should render children', () => {
    const wrapper = shallow(
      <AppHeader>
        <img className="child" />
        <h1>Jedi List</h1>
      </AppHeader>
    )

    expect(wrapper.find('img').hasClass("child")).toEqual(true)
    expect(wrapper.find('h1').text()).toEqual("Jedi List")
    expect(wrapper.html()).toMatchSnapshot()
  })
})
