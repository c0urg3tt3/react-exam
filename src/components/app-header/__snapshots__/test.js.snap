// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[component] AppHeader should render child 1`] = `"<header class=\\"app-header\\"><h1>Jedi List</h1></header>"`;

exports[`[component] AppHeader should render children 1`] = `"<header class=\\"app-header\\"><img class=\\"child\\"/><h1>Jedi List</h1></header>"`;

exports[`[component] AppHeader should render component 1`] = `"<header class=\\"app-header\\"></header>"`;
